src/mappings/vault.ts

```diff
@@ -112,14 +118,27 @@ function handlePoolJoined(event: PoolBalanceChanged): void {
   for (let i: i32 = 0; i < tokenAddresses.length; i++) {
     let tokenAddress: Address = Address.fromString(tokenAddresses[i].toHexString());
     let poolToken = loadPoolToken(poolId, tokenAddress);
+    let token = getToken(tokenAddress);
+    let tokenSnapshot = getTokenSnapshot(tokenAddress, event);
+
     // adding initial liquidity
     if (poolToken == null) {
       throw new Error('poolToken not found');
     }
     let tokenAmountIn = tokenToDecimal(amounts[i], poolToken.decimals);
     let newAmount = poolToken.balance.plus(tokenAmountIn);
+    let tokenAmountInUSD = valueInUSD(tokenAmountIn, tokenAddress);
+
+    token.totalBalanceNotional = token.totalBalanceNotional.plus(tokenAmountIn);
+    token.totalBalanceUSD = token.totalBalanceUSD.plus(tokenAmountInUSD);
+
+    tokenSnapshot.balanceNotional = tokenSnapshot.balanceNotional.plus(tokenAmountIn);
+    tokenSnapshot.balanceUSD = tokenSnapshot.balanceUSD.plus(tokenAmountInUSD);
+
     poolToken.balance = newAmount;
     poolToken.save();
+    token.save();
+    tokenSnapshot.save();
```

Comment:
I tend to prefer keeping entities "open" for as short a time as possible so it's easy to see when we're using them and how they're being updated. Would you be happy to update to something like this style?

```suggestion
    
    // adding initial liquidity
    if (poolToken == null) {
      throw new Error('poolToken not found');
    }
    let tokenAmountIn = tokenToDecimal(amounts[i], poolToken.decimals);
    let newAmount = poolToken.balance.plus(tokenAmountIn);
    let tokenAmountInUSD = valueInUSD(tokenAmountIn, tokenAddress);

    poolToken.balance = newAmount;
    poolToken.save();

    let token = getToken(tokenAddress);
    token.totalBalanceNotional = token.totalBalanceNotional.plus(tokenAmountIn);
    token.totalBalanceUSD = token.totalBalanceUSD.plus(tokenAmountInUSD);
    token.save();

    let tokenSnapshot = getTokenSnapshot(tokenAddress, event);
    tokenSnapshot.balanceNotional = tokenSnapshot.balanceNotional.plus(tokenAmountIn);
    tokenSnapshot.balanceUSD = tokenSnapshot.balanceUSD.plus(tokenAmountInUSD);
    tokenSnapshot.save();
```

Reply:
sounds good

Reply:
Cool, can you make these changes?

Reply:
yeah, was a bit busy yesterday with front-end. this is resolved